datalad/core/local/create.py

```diff
@@ -339,6 +339,23 @@ def __call__(
             yield res
             return
 
+        # Check if specified cfg_proc(s) can be discovered, storing
+        # the results so they can be used when the time comes to run
+        # the procedure. If a procedure cannot be found, raise an
+        # error to prevent creating the dataset.
+        cfg_proc_specs = []
+        for cfg_proc_ in cfg_proc:
+            for discovered_proc in tbds.run_procedure(
+                    discover=True,
+                    result_renderer='disabled',
+                    return_type='generator'):
```

Comment:
If I read this correctly, then this will call `run_procedure` once for each given config procedure name. This seems wasteful, as the result of each call will be identical. Why not call it once, and then search for each given `cfg_proc`?

Reply:
I assumed there won't be many cases when multiple cfg_procs are given, but indeed I wasn't sure about this part. Probably better to create a list once than a generator (possibly) many times. I'm changing according to your suggestion.